### PR TITLE
add docker file and readme

### PR DESCRIPTION
**Pull request by VincyZhang**
## Type of Change

Add docker file for end-users to build docker images

### CODE REVIEW

**Comment by VincyZhang** [comment]
@airMeng @luoyu-intel  what's this issue? using GCC/G++ 11

image

**Comment by VincyZhang** [comment]
GCC Version (G++ is same as GCC)

Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/11/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none:amdgcn-amdhsa
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 11.4.0-1ubuntu1~22.04' --with-bugurl=file:///usr/share/d                                                     oc/gcc-11/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++,m2 --prefix=/usr --with-gcc-major-ver                                                     sion-only --program-suffix=-11 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir                                                     =/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-                                                     clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-o                                                     bject --disable-vtable-verify --enable-plugin --enable-default-pie --with-system-zlib --enable-libphobos-checking=relea                                                     se --with-target-system-zlib=auto --enable-objc-gc=auto --enable-multiarch --disable-werror --enable-cet --with-arch-32                                                     =i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=n                                                     vptx-none=/build/gcc-11-XeT9lY/gcc-11-11.4.0/debian/tmp-nvptx/usr,amdgcn-amdhsa=/build/gcc-11-XeT9lY/gcc-11-11.4.0/debi                                                     an/tmp-gcn/usr --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --targe                                                     t=x86_64-linux-gnu --with-build-config=bootstrap-lto-lean --enable-link-serialization=2
Thread model: posix
Supported LTO compression algorithms: zlib zstd
gcc version 11.4.0 (Ubuntu 11.4.0-1ubuntu1~22.04)

**Comment by airMeng** [comment]
have you copied bestla?

**Comment by VincyZhang** [comment]
> have you copied bestla?

how and why? I just clone the repo and init the submodules, then run cmake/make

File: docker/DockerFile [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    cmake \
+    git \
+    gcc g++ make
+
+RUN mkdir -p /neural_speed
+WORKDIR /neural_speed
+COPY . /neural_speed
+
+RUN pip install cmake ninja psutil && \
+    cd /neural_speed/neural_speed && \
```
Comment (airMeng) [inline docker/DockerFile:44]: should be 
```suggestion
    cd /neural_speed && \
```

**Review by airMeng — APPROVED** [review]

**Review by luoyu-intel — APPROVED** [review]

**Review by kevinintel — APPROVED** [review]

**Pull request merged**